### PR TITLE
CATROID-1093 REFACTOR ListProjectFilesTask to Kotlin

### PR DESCRIPTION
**Pull request by hofcle78**
https://jira.catrob.at/browse/CATROID-1093

### Your checklist for this pull request
Please review the [contributing guidelines](https://github.com/Catrobat/Catroid/blob/develop/README.md) and [wiki pages](https://github.com/Catrobat/Catroid/wiki/) of this repository.

- [x] Include the name of the Jira ticket in the PR’s title
- [x] Include a summary of the changes plus the relevant context
- [x] Choose the proper base branch (*develop*)
- [x] Confirm that the changes follow the project’s coding guidelines
- [x] Verify that the changes generate no compiler or linter warnings
- [x] Perform a self-review of the changes
- [x] Verify to commit no other files than the intentionally changed ones
- [x] Include reasonable and readable tests verifying the added or changed behavior
- [x] Confirm that new and existing unit tests pass locally
- [x] Check that the commits’ message style matches the [project’s guideline](https://github.com/Catrobat/Catroid/wiki/Commit-Message-Guidelines)
- [x] Stick to the project’s gitflow workflow
- [x] Verify that your changes do not have any conflicts with the base branch
- [x] After the PR, verify that all CI checks have passed
- [ ] Post a message in the *catroid-stage* or *catroid-ide* [Slack channel](https://catrobat.slack.com) and ask for a code reviewer

### CODE REVIEW

File: catroid/src/main/java/org/catrobat/catroid/ui/filepicker/FilePickerActivity.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+        findViewById<View>(R.id.recycler_view).visibility = if (show) View.GONE else View.VISIBLE
+    }
+
+    private fun getFiles() {
+        object : RequiresPermissionTask(PERMISSIONS_REQUEST_IMPORT_FROM_EXTERNAL_STORAGE,
+            listOf(permission.READ_EXTERNAL_STORAGE),
+            R.string.runtime_permission_general
+        ) {
+            override fun task() =
+                ProjectFileLister(this@FilePickerActivity).listProjectFilesAsync(getStorageRoots())
```
Comment (frit96) [inline catroid/src/main/java/org/catrobat/catroid/ui/filepicker/FilePickerActivity.kt:76]:
```suggestion
                ProjectFileLister().listProjectFilesAsync(getStorageRoots(), this@FilePickerActivity)
```

**Review by frit96 — COMMENTED** [review]
Overall it looks very good just add the small changes I requested.

**Comment by frit96** [comment]
Thanks looks good now!

**Review by petbuer — CHANGES_REQUESTED** [review]
Please have a look at my comments.

**Review by frit96 — APPROVED** [review]

**Comment by frit96** [comment]
@hofcle78 please resolve the conflicts and rebase

**Comment by hofcle78** [comment]
Done. :+1:

**Comment by reichli** [comment]
closed due to inactivity

**Pull request closed without merging**